File: backend/tela-api/src/model/entities/produto.py
from sqlalchemy import Column, String, DECIMAL, INTEGER,DATETIME, Boolean, ForeignKey
from sqlalchemy.orm import relationship

from .base_entity import BaseEntity


class Produto(BaseEntity):
    __tablename__ = 'produtos'

    codigo = Column(String(30), unique=True, nullable=False)
    nome = Column(String(250), default='')
    cod_barras = Column(String(14), default='')
    referencia = Column(String(250), default='')
    observacao = Column(String(510), default='')

    pr_venda_vista = Column(DECIMAL(10, 2), default=0.01)
    pr_venda_prazo = Column(DECIMAL(10, 2), default=0.01)
    pr_custo = Column(DECIMAL(12,3))
    outros = Column(DECIMAL(12,3))

    estoque = Column(DECIMAL(12, 3), default=0.0)
    estoque_minimo = Column(DECIMAL(10, 2))

    unidade = Column(String(6), ForeignKey('unidades.unid'), nullable=False)
    qtd_embalagem = Column(DECIMAL(10,2), default=0.0)
    setor_id = Column(INTEGER, ForeignKey('setores.id'), nullable=False)
    setor = relationship('Setor')

    ultima_compra = Column(DATETIME)
    ultima_venda = Column(DATETIME)
    ativo = Column(Boolean, default=True, nullable=False)

File: frontend/tela-retaguarda/src/app/shared/models/entity/produto.ts
import { Setor } from 'src/app/shared/models/entity/setor';
import { Unidade } from 'src/app/shared/models/entity/unidade';
import { BaseEntity } from './base';

export  interface Produto extends BaseEntity {
  codigo?:string
  nome?:string
  cod_barras:number
  referencia?:string
  observacao?:string

  pr_venda_vista?:number
  pr_venda_prazo?:number

  estoque?:number
  estoque_minimo?:number
  pr_custo?:number
  outros?:number

  qtd_embalagem:number

  unidade?:Unidade
  setor?:Setor

  ultima_compra?:Date
  ultima_venda?:Date
  ativo?:boolean
}
